### PR TITLE
Better highlighter

### PR DESCRIPTION
**Pull request by dblanovschi**
Everything has colors now.

### CODE REVIEW

**Comment by Mcat12** [comment]
Thanks for working this, though I don't think this is the right approach... This supports primitive types like integers and `str`, but does not support non-primitive types like `Vec` or user-defined types. I wrote down some thoughts in #12 for solving the general case of this problem, basically injecting Rust whenever we reference a Rust type. Adding that support would implicitly add the highlighting of primitives like this PR does, but it would also provide reference resolution, autocomplete, and everything else that the Rust plugin provides for types.

**Comment by dblanovschi** [comment]
> Adding that support would implicitly add the highlighting of primitives like this PR does, but it would also provide reference resolution, autocomplete, and everything else that the Rust plugin provides for types.

Yes, you're right, but aside those what do you think about the highlighting of everything else?

**Comment by Mcat12** [comment]
I didn't run the code until now so I did miss that, but I do like the highlighting done on the rest of the grammar. So how about remove the Rust type highlighting and keep the rest (nonterminals, etc).

File: src/main/kotlin/com/mdrobnak/lalrpop/resolve/LpPathIdReference.kt [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+import com.intellij.openapi.util.TextRange
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiReferenceBase
+import com.intellij.psi.util.PsiTreeUtil
+import com.mdrobnak.lalrpop.psi.LpElementFactory
+import com.mdrobnak.lalrpop.psi.LpNonterminal
+import com.mdrobnak.lalrpop.psi.LpPathId
+
+class LpPathIdReference(element: LpPathId) :
```
Comment (Mcat12) [inline src/main/kotlin/com/mdrobnak/lalrpop/resolve/LpPathIdReference.kt:11]: Why is this added? Path IDs should reference Rust types, not nonterminals.

Comment (dblanovschi) [reply src/main/kotlin/com/mdrobnak/lalrpop/resolve/LpPathIdReference.kt:11]: In the case of:
```
NonTerminal<Rule>: Type<Rule> = {}
```

The second `Rule` should references the first

Comment (Mcat12) [reply src/main/kotlin/com/mdrobnak/lalrpop/resolve/LpPathIdReference.kt:11]: It might not be entirely useless, but I'm unsure how it will respond once we inject Rust into the nonterminal type / type refs (#12).

Comment (dblanovschi) [reply src/main/kotlin/com/mdrobnak/lalrpop/resolve/LpPathIdReference.kt:11]: So... should I remove it?

Comment (Mcat12) [reply src/main/kotlin/com/mdrobnak/lalrpop/resolve/LpPathIdReference.kt:11]: Sorry, been a bit busy so haven't been able to look at this more until now. We can leave it in until we resolve #12, at which point this may be removed (or it stays helpful for resolving generic types).

**Review by Mcat12 — APPROVED** [review]
